sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py

```diff
@@ -186,12 +188,48 @@ def managed(self):
 
 
 class KeyVaultKey(object):
-    """A key's attributes and cryptographic material"""
+    """A key's attributes and cryptographic material.
+
+    :param str key_id:
+        Key Vault's identifier for the key. Typically a URI, e.g. https://myvault.vault.azure.net/keys/my-key/version
+    :param jwk:
+        The key's cryptographic material as a JSON Web Key (https://tools.ietf.org/html/rfc7517). This may be provided
+        as a dictionary or keyword arguments. See :class:`~azure.keyvault.keys.models.JsonWebKey` for field names.
+
+    Providing cryptographic material as keyword arguments:
+
+    .. code-block:: python
+
+        from azure.keyvault.keys.models import KeyVaultKey
+
+        key_id = 'https://myvault.vault.azure.net/keys/my-key/my-key-version'
+        key_bytes = os.urandom(32)
+        key = KeyVaultKey(key_id, k=key_bytes, kty='oct', key_ops=['unwrapKey', 'wrapKey'])
+
+    Providing cryptographic material as a dictionary:
+
+    .. code-block:: python
+
+        from azure.keyvault.keys.models import KeyVaultKey
+
+        key_id = 'https://myvault.vault.azure.net/keys/my-key/my-key-version'
+        key_bytes = os.urandom(32)
+        jwk = {'k': key_bytes, 'kty': 'oct', 'key_ops': ['unwrapKey', 'wrapKey']}
+        key = KeyVaultKey(key_id, jwk=jwk)
+
+    """
 
-    def __init__(self, properties, key_material):
-        # type: (KeyProperties, _models.JsonWebKey, **Any) -> None
-        self._properties = properties
-        self._key_material = key_material
+    def __init__(self, key_id, jwk=None, **kwargs):
+        # type: (str, Optional[dict], **Any) -> None
+        self._properties = kwargs.pop("properties", None) or KeyProperties(key_id, **kwargs)
+        if isinstance(jwk, dict):
```

Comment:
I'm good with checking this in now, but checking for `dict` specifically is problematic. Please open an issue to track supporting other mapping types - assuming that the user wanting to provide one is even a remote possibility.

Reply:
#8168